src/test/scala/org/hammerlab/guacamole/reads/ReadSetSuite.scala

```diff
@@ -20,36 +20,75 @@ package org.hammerlab.guacamole.reads
 
 import org.apache.parquet.hadoop.metadata.CompressionCodecName
 import org.bdgenomics.adam.rdd.read.AlignmentRecordRDDFunctions
-import org.bdgenomics.adam.rdd.{ ADAMSaveAnyArgs, ADAMContext }
-import org.hammerlab.guacamole.reference.ReferenceBroadcast
-import org.hammerlab.guacamole.{ LociSet }
+import org.bdgenomics.adam.rdd.{ADAMContext, ADAMSaveAnyArgs}
+import org.hammerlab.guacamole.LociSet
 import org.hammerlab.guacamole.reads.Read.InputFilters
-import org.hammerlab.guacamole.util.{ TestUtil, GuacFunSuite }
+import org.hammerlab.guacamole.reference.ReferenceBroadcast
+import org.hammerlab.guacamole.util.{GuacFunSuite, TestUtil}
 import org.scalatest.Matchers
 
 class ReadSetSuite extends GuacFunSuite with Matchers {
 
   def chr22Fasta = ReferenceBroadcast.readFasta(TestUtil.testDataPath("chr22.fa.gz"), sc)
 
+  case class LazyMessage(msg: () ⇒ String) {
+    override def toString: String = msg()
+  }
+
   sparkTest("using different bam reading APIs on sam/bam files should give identical results") {
     def check(paths: Seq[String], filter: InputFilters): Unit = {
       withClue("using filter %s: ".format(filter)) {
-        val configs = Read.ReadLoadingConfig.BamReaderAPI.values.map(api => Read.ReadLoadingConfig(bamReaderAPI = api)).toSeq
-        val standard = TestUtil.loadReads(sc, paths(0), filter, config = configs(0), reference = chr22Fasta).reads.collect
-        configs.foreach(config => {
-          paths.foreach(path => {
-            withClue("file %s with config %s vs standard %s with config %s:\n".format(path, config, paths(0), configs(0))) {
-              val result = TestUtil.loadReads(sc, path, filter, config = config, reference = chr22Fasta).reads.collect
-              if (result.toSet != standard.toSet) {
-                val missing = standard.filter(!result.contains(_))
-                assert(missing.isEmpty, "Missing reads: %s".format(missing.map(_.toString).mkString("\n\t")))
-                val extra = result.filter(!standard.contains(_))
-                assert(extra.isEmpty, "Extra reads:\n\t%s".format(extra.map(_.toString).mkString("\n\t")))
-                assert(false, "shouldn't get here")
-              }
-            }
-          })
-        })
+
+        val firstPath = paths.head
+
+        val configs =
+          Read.ReadLoadingConfig.BamReaderAPI.values
+            .map(api => Read.ReadLoadingConfig(bamReaderAPI = api))
+
+        val firstConfig = configs.head
+
+        val standard =
+          TestUtil.loadReads(
+            sc,
+            paths.head,
+            filter,
+            config = configs.head,
+            reference = chr22Fasta
+          ).reads.collect
+
+        for {
+          config ← configs
```

Comment:
> They will look odd to someone new to scala (even someone not so new, like me)

Seems to me they would look about as weird to someone new to scala as their ascii-art equivalents; they look weird to you because you're not new to Scala and are used to the old way.

> There are lots of other non-single-character syntax tokens like 'val', so I don't see a problem

If there were a symbol that the entire toolchain was supporting instead of `val` I would be interested in using it / allowing it in the codebase.

In any case I removed them here and force-pushed.